rp2040-hal/src/i2c/peripheral.rs

```diff
@@ -94,120 +112,209 @@ where
             // hold scl when fifo's full
             w.rx_fifo_full_hld_ctrl().enabled();
             w.ic_restart_en().enabled();
-            w.tx_empty_ctrl().enabled()
+            w.ic_10bitaddr_slave().variant(A::BIT_ADDR_S);
+            //w.tx_empty_ctrl().enabled();
+            w
         });
 
         // Clear FIFO threshold
+        // SAFETY: Only address by this function. The field is 8bit long. 0 is a valid value.
         i2c.ic_tx_tl.write(|w| unsafe { w.tx_tl().bits(0) });
         i2c.ic_rx_tl.write(|w| unsafe { w.rx_tl().bits(0) });
 
+        i2c.ic_clr_intr.read();
+
+        let mut me = Self {
+            i2c,
+            pins: (sda_pin, scl_pin),
+            mode: Peripheral { state: State::Idle },
+        };
+        me.unmask_intr();
         // Enable I2C block
-        i2c.ic_enable.write(|w| w.enable().enabled());
-
-        I2CPeripheralEventIterator {
-            i2c: Self {
-                i2c,
-                pins: (sda_pin, scl_pin),
-                mode: PhantomData,
-            },
-            state: State::Idle,
-        }
+        me.i2c.ic_enable.write(|w| w.enable().enabled());
+
+        me
     }
 }
 
-impl<T: Deref<Target = I2CBlock>, PINS> I2CPeripheralEventIterator<T, PINS> {
+/// SAFETY: Takes a non-mutable reference to RegisterBlock but mutates its `ic_intr_mask` register.
+unsafe fn unmask_intr(i2c: &RegisterBlock) {
+    // SAFETY: 0 is a valid value meaning all irq masked.
+    i2c.ic_intr_mask.write_with_zero(|w| {
+        // Only keep these IRQ enabled.
+        w.m_start_det()
+            .disabled()
```

Comment:
The implicit double negation makes this difficult to understand.
Am I right that `disabled` means that a 1 is written to the register bit, meaning that the masking is disabled, ie. the interrupt is active?

Reply:
Yes, `disabled` means the mask is disabled, so the interrupt is `not` masked which is done by setting the corresponding bit in the field.